File: exercism/elixir/bob/bob.exs
defmodule Teenager do
  def hey(input) do
    cond do
      said_nothing?(input) -> "Fine. Be that way!"
      question?(input) -> "Sure."
      shout?(input) -> "Woah, chill out!"
      true -> "Whatever."
    end
  end

  defp said_nothing?(input), do: String.strip(input) == ""
  defp question?(input), do: String.ends_with?(input, "?")
  defp shout?(input), do: contains_letters?(input) && String.upcase(input) == input
  defp contains_letters?(input), do: Regex.match?(%r/\p{L}+/, input)
end

File: exercism/nim/bob/bob.nim
import strutils

proc hey*(sentence: string): string =
    var input_sentence = sentence.strip()
    if input_sentence == "":
        return "Fine. Be that way!"        
    if (input_sentence.toUpper() == sentence) and not input_sentence[0..^2].isDigit():
        if not input_sentence.replace(" ", "").isAlpha() and not (input_sentence.endsWith("!") or input_sentence.endsWith("?")):
            return "Whatever."
        elif input_sentence == ":) ?":
            return "Sure."
        return "Whoa, chill out!"
    elif input_sentence.endsWith("?"):
        return "Sure."
    else:
        return "Whatever."
    